external-crates/move/crates/move-binary-format/src/compatibility.rs

```diff
@@ -349,6 +416,50 @@ impl InclusionCheck {
             };
         }
 
+        // Enum checks
+        for (name, old_enum) in &old_module.enums {
+            match new_module.enums.get(name) {
+                Some(new_enum) => {
+                    if old_enum.abilities != new_enum.abilities {
+                        return err;
+                    }
+                    if old_enum.type_parameters != new_enum.type_parameters {
+                        return err;
+                    }
+                    if old_enum.variants.len() > new_enum.variants.len() {
+                        return err;
+                    }
+
+                    // NB: In the future when we allow adding new variants to enums in subset mode
+                    // remove this if statement. This check is somewhat redundant with the one
+                    // below, the one below should be kept when we allow adding variants in subset
+                    // mode.
+                    if old_enum.variants.len() != new_enum.variants.len() {
+                        return err;
+                    }
+
+                    if self == &Self::Equal && old_enum.variants.len() != new_enum.variants.len() {
```

Comment:
What are the other settings here for `Self`?

Reply:
The other setting would be `Subset` -- which would be additive-only upgrades. Note that in a world where we allowed adding variants it would (IMO) be allowed under that setting.